docs/pages/project/contributing/contributing-error.md

```diff
@@ -9,27 +9,121 @@ category: contributing
 ---
 
 Meshery pervasively uses MeshKit as a golang and service mesh management-specific library in all of its components. MeshKit helps populate error messages with a uniform and useful set of informative attributes. 
-The below are the listed attributes:
+
+To help with creating error codes, MeshKit contains a tool that analyzes, verifies and updates error codes in Meshery source code trees. It extracts error details into a file that can be used for publishing all error code references on the at Meshery [error codes reference page](https://docs.meshery.io/reference/error-codes). The objective to create this was to avoid centralized handling of error codes and automating everything
+
+In order to create a Meshery error object, you will need to create a custom wrapper object for the native golang error. This can be done from the <a href="https://github.com/layer5io/meshkit/tree/master/errors">MeshKit Error</a> package. 
+
+
+## Some rules for making Errors codes
+
+- Errors names and codes are namespaced to components, i.e. they need to be unique within a component, which is verified by this tool.
+
+- Errors are not to be reused across components and modules.
+
+- Codes carry no meaning, as e.g. HTTP status codes do.
+
+- In the code, create string var's or const's with names starting with Err[A-Z] and ending in Code, e.g. 'ErrApplyManifestCode'.
+
+- Set the value to any string, like "replace_me" (no convention here), e.g. ErrApplyManifestCode = "replace_me".
+
+- Error codes are not to be set as integer 
+
+- CI will take care of updating Error codes from a string to an integer.
+
+- Using errors.NewDefault(...) is deprecated. This tool emits a warning if this is detected.
+
+- Use errors.New(...) from MeshKit to create actual errors with all the details.
+  This is often done in a factory function. It is important that the error code variable is used here, not a literal.
+  Specify detailed descriptions, probable causes, and remedies. They need to be string literals, call expressions are ignored.
```

Comment:
And I think it can be added that the `first letter of the every error description need to capitalized`